packages/teleport/src/dashboard/People/Users/UserDelete/UserDelete.tsx

```diff
@@ -74,8 +74,8 @@ export function UserDelete({
 function useDialog(props: Props) {
   const { attempt, run } = useAttemptNext();
   function onDelete() {
-    return run(() => props.onDelete(props.username)).then(() =>
-      props.onClose()
+    return run(() =>
+      props.onDelete(props.username).then(() => props.onClose())
```

Comment:
it will through warnings in the console about updating unmounted component. So in this case you need to use `setAttmpt({ status: 'XX'})` explicitly. See how it's done in UserAddEdit hook.

Reply:
revised, and tested in UI